mllib/src/main/scala/org/apache/spark/ml/feature/VectorIndexer.scala

```diff
@@ -248,6 +252,16 @@ class VectorIndexerModel private[ml] (
     val categoryMaps: Map[Int, Map[Double, Int]])
   extends Model[VectorIndexerModel] with VectorIndexerParams {
 
+  /** Java-friendly version of [[categoryMaps]] */
+  def javaCategoryMaps: JMap[java.lang.Integer, JMap[java.lang.Double, java.lang.Integer]] = {
+    categoryMaps.map { case (feature, catMap) =>
```

Comment:
With `JavaConverters`, this could be simplified to

~~~scala
categoryMaps.mapValues(_.asJava).asJava.asInstanceOf[JMap[java.lang.Integer, JMap[java.lang.Double, java.lang.Integer]]]
~~~